File: v1/src/main/java/com/api/v1/customers/domain/CustomerRepository.java
package com.api.v1.customers.domain;

import java.util.Optional;

import org.springframework.data.mongodb.repository.MongoRepository;
import org.springframework.data.mongodb.repository.Query;

public interface CustomerRepository extends MongoRepository<Customer, String> {

    @Query("{ 'person.sin': ?0 }")
    Optional<Customer> findBySIN(String sin);

    @Query("{ 'person.email': ?0 }")
    Optional<Customer> findByEmail(String email);
}

File: v2/src/main/kotlin/com/api/v2/customers/domain/CustomerRepository.kt
package com.api.v2.customers.domain

import org.springframework.data.mongodb.repository.MongoRepository
import org.springframework.data.mongodb.repository.Query

interface CustomerRepository: MongoRepository<Customer, String> {

    @Query("{ 'person.sin': ?0 }")
    suspend fun findBySIN(sin: String)

    @Query("{ 'person.email': ?0 }")
    suspend fun findByEmail(email: String)
}